### PR TITLE
intg tests for 'any_errors_fatal'

### PR DESCRIPTION
**Pull request by alikins**
##### SUMMARY

Based on https://github.com/ansible/ansible/issues/18602 
https://github.com/ansible/ansible/issues/22924

##### ISSUE TYPE

 - Bugfix Pull Request

##### COMPONENT NAME

test/integration/targets/any_error_fatal

##### ANSIBLE VERSION

```

ansible 2.4.0 (any_errors_fatal_intg_tests 6e8c534f0e) last updated 2017/06/16 16:01:49 (GMT -400)
  config file = /home/adrian/src/ansible-setup/ansible.cfg
  configured module search path = [u'/home/adrian/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /home/adrian/src/ansible/lib/ansible
  executable location = /home/adrian/src/ansible/bin/ansible
  python version = 2.7.13 (default, May 10 2017, 20:04:28) [GCC 6.3.1 20161221 (Red Hat 6.3.1-1)]
```

##### ADDITIONAL INFORMATION

```

```

### CODE REVIEW

File: test/integration/targets/any_errors_fatal/always_block.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+---
+- hosts: testhost
+  gather_facts: false
+  any_errors_fatal: true
+  tasks:
+    - block:
+        - name: initial block debug
+          debug: msg='any_errors_fatal_block,  i execute normally'
+
+        - name: any_errors_fatal, initial block, bin/fasel to simulate failure
```
Comment (gundalow) [inline test/integration/targets/any_errors_fatal/always_block.yml]: typo bin/false

File: test/integration/targets/any_errors_fatal/always_block.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+          command: /bin/false
+
+        - name: after a task that fails I should never execute
+          debug:
+            msg: 'any_errors_fatal_block_post_fail ... i never execute, cause ERROR!'
+      rescue:
+        - name: any_errors_fatal_rescue_block debug
+          debug: msg='any_errors_fatal_rescue_block_start ... I caught an error'
+
+        - name: any_errors_fatal in rescue block, using bin/flase to simulate error
```
Comment (gundalow) [inline test/integration/targets/any_errors_fatal/always_block.yml]: typo bin/false

**Review by gundalow — DISMISSED** [review]
Test is failing on some OS

**Comment by mattclay** [comment]
CI failure in the new integration tests. Here's one of the errors:

```
{
"failed": true, 
"msg": "the field 'args' has an invalid value, which appears to include a variable that is undefined.The error was: 'test' is undefined\n\nThe error appears to have been in '/root/ansible/test/integration/targets/any_errors_fatal/test_fatal.yml': line 7, column 3, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n\n- shell: \"echo {{ test }}\"\n  ^ here\nWe could be wrong, but this one looks like it might be an issue with\nmissing quotes.  Always quote template expression brackets when they\nstart a value. For instance:\n\n    with_items:\n      - {{ foo }}\n\nShould be written as:\n\n    with_items:\n      - \"{{ foo }}\"\n\nexception type: <class 'ansible.errors.AnsibleUndefinedVariable'>\nexception: 'test' is undefined"
}
```

**Comment by alikins** [comment]
@mattclay Not sure why these tests are failing. The runme.sh expects the playbooks to fail in different ways, and it seems to exit with a 0 return code.

The runme.sh does do some odd things to echo the playbook output though. Could that be flagging something as a failure? 

Running tests via ansible-test 'ansible-test integration --docker centos7 -vvvv any_errors_fatal' doesnt fail
and exits with 'Command exited with status 0 after 0.778028964996 seconds.' It looks like CI does the same. Is something parsing the output?

ansible-test output from above is at https://gist.github.com/alikins/ddc6da003654fca669511a4f2e01736c

**Comment by mattclay** [comment]
@alikins Tasks that are supposed to fail and are not marked `ignore_errors: yes` will need to have `EXPECTED FAILURE` in their task name to inform CI that those tasks should be treated as passing.

Example: https://github.com/ansible/ansible/blob/devel/test/integration/targets/blocks/main.yml#L18

**Review by nitzmahone — APPROVED** [review]
This looks great- sorely needed. Bonus points for adding include/import_tasks coverage too, but let's not hold up merging for that.

**Comment by nitzmahone** [comment]
rebuild_merge

**Comment by alikins** [comment]
@nitzmahone I don't have any include_task/import_task tests but https://github.com/ansible/ansible/pull/30714 has some include_role/import_role tests.  30714 failed against devel last I check, but the errors it hits (include_role not being json serializable) have been fixed I believe. I'll rebase and update 30714

**Comment by mattclay** [comment]
GH says there are issues with rebasing this branch. That may prevent it from passing CI.

**Pull request merged**